File: 1803-average-waiting-time/average-waiting-time.java
class Solution {
    public double averageWaitingTime(int[][] customers) {
        double ans = 0;
        double cnt = 0;
        int chef = 0;
        for(int i=0; i<customers.length; i++) {
            if(chef < customers[i][0]) chef = customers[i][0];
            chef+=customers[i][1];
            ans+=(chef - customers[i][0]);
            cnt++;
        } return ans/cnt;
    }
}

File: 1803-average-waiting-time/average-waiting-time.cpp
class Solution {
public:
    double averageWaitingTime(vector<vector<int>>& customers) {
        int chef = 0;
        double ans = 0,cnt=0;
        // cout << customers[0].size();
        for(int i=0; i<customers.size(); i++) {
            if(chef < customers[i][0]) {
                chef = customers[i][0];
                cout <<"arriving : " <<chef << endl;
            }
            chef+=customers[i][1];
            cout << chef << endl;
            ans+=(chef - customers[i][0]);
            cout <<"watiting : "<<ans<<endl;
            cnt++;
        }
        return ans/cnt;
    }
};